File: backend/src/main/java/unpsjb/labprog/backend/model/HorarioEstacionamiento.java
package unpsjb.labprog.backend.model;

import java.util.Collection;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class HorarioEstacionamiento {
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    private String nombre;

    // Solo día y mes, sin año
    private String fechaInicio;
    private String fechaFin;

    @OneToMany(cascade = CascadeType.ALL)
    private Collection<HoraInicioHoraFin> horariosDelDia;
    
}

File: frontend/cli/src/app/horario.estacionamiento/horario.estacionamiento.ts
import { HoraInicioHoraFin } from "./hora.inicio.hora.fin";

export interface HorarioEstacionamiento {
    id: number;
    nombre: string;

    fechaInicio: string;  // Por ejemplo, "05-01" para 1 de mayo
    fechaFin: string;     // Por ejemplo, "09-30" para 30 de septiembre

    horariosDelDia : HoraInicioHoraFin[];
}
